R/hs_PCA.R

```diff
@@ -0,0 +1,29 @@
+#' Run PCA on a hyperspec object
+#'
+#' @param hs.x hyperSpec object
+#' @param ... Parameters to pass on to prcomp().
+#' @importFrom stats prcomp
+#' @examples
+#' ## Short example
+#'
+#' # Load hyperSpec object
+#' data("hs_example")
+#'
+#' # Convert to MassSpectrum object
+#' hs.x.proc <- hs_preprocess(hs_example)
+#'
+#' df.PCA <- hs_PCA(hs.x.proc)
+#' @export
+
+
+hs_PCA <- function(hs.x, ...) {
+  if (is.null(rownames(hs.x@data$spc)))
+    spectra_ids <-
+      hs.x@data$filename
+  else
+    spectra_ids <- rownames(hs.x@data$spc)
+  hs.PCA <- prcomp(x = hs.x@data$spc, ...)
```

Comment:
if feature extraction is the idea, princomp could be desirable, I am not sure here what the aim of the PCA is.

Reply:
Primarily for dim reduction/visualization purposes - can add princomp if needed in future